x-pack/plugins/security_solution/public/actions/add_to_timeline/index.ts

```diff
@@ -5,4 +5,5 @@
  * 2.0.
  */
 
-export { AddToTimelineAction } from './add_to_timeline_action';
+export { createAddToTimelineAction } from './add_to_timeline';
+export { EmbeddableAddToTimelineAction } from './embeddable_add_to_timeline_action';
```

Comment:
If we prepend the "target" of the actions in the name we should also name the first one as "cell_actions_add_to_timeline" or something.
But probably it would be cleaner to separate the actions into different directories:

- security_solution/public/actions/lens_embeddable/
  - add_to_timeline
  - copy_to_clipboard
- security_solution/public/actions/default_cell_actions/
  - ...
- security_solution/public/actions/timeline_cell_actions/
  - ...

What do you think?

Reply:
It makes sense to add `cell_actions` to the file name. But following the suggested folder organization would make sharing logic between the three implementations of add_to_timeline more difficult. Where could we place the `add_to_timeline` common code?

Reply:
I see, what about this:

- security_solution/public/actions/add_to_timeline/
  - common_things.ts
  - lens_embeddable/
    - add_to_timeline.ts
    - add_to_timeline.test.ts 
    - other_things_related_to_this_specific_action.ts
    - index.ts (?)
  - default_cell_actions/
    - ...
  - timeline_cell_actions/
    - ...

Reply:
I implemented your idea with a slight change. I removed embeddable and CellActions from the file names because it was getting huge. So instead, I am using only the place where the action is used (timeline, default, lens).


* security_solution/public/actions/add_to_timeline/
  * index.ts `// exports renamed functions for all lens, default, and timeline folders`
  * common_things.ts
  * lens/
    * add_to_timeline.ts
    * add_to_timeline.test.ts
    * other_things_related_to_this_specific_action.ts
  * default/
    * ...
  * timeline/
    * ...

It looks cleaner and more organized than what I had implemented. Let me know what you think.

Reply:
Awesome! thanks